packages/components/src/icon/index.tsx

```diff
@@ -1,50 +1,36 @@
 /**
- * External dependencies
+ * Forked from `@wordpress/components`
+ *
+ * Only forked for CSS collision safety, in case styles or classnames are added upstream.
+ *
+ * - Removed Gridicons support (non-critical).
  */
 
-/**
- * WordPress dependencies
- */
 import { cloneElement, createElement, isValidElement } from '@wordpress/element';
 import { SVG } from '@wordpress/primitives';
+import type { ComponentType } from 'react';
 
-/**
- * Internal dependencies
- */
-import Dashicon from '../dashicon';
-import type { IconKey as DashiconIconKey } from '../dashicon/types';
-import type { ComponentType, HTMLProps, SVGProps } from 'react';
-
-export type IconType =
-	| DashiconIconKey
+type IconType =
 	| ComponentType< { size?: number } >
 	| ( ( props: { size?: number } ) => JSX.Element )
 	| JSX.Element;
 
-type AdditionalProps< T > = T extends ComponentType< infer U >
-	? U
-	: T extends DashiconIconKey
-	? SVGProps< SVGSVGElement >
-	: {};
+type AdditionalProps< T > = T extends ComponentType< infer U > ? U : Record< string, unknown >;
 
-export type Props = {
+type Props = {
 	/**
 	 * The icon to render. In most cases, you should use an icon from
 	 * [the `@wordpress/icons` package](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library).
 	 *
-	 * Other supported values are: component instances, functions,
-	 * [Dashicons](https://developer.wordpress.org/resource/dashicons/)
-	 * (specified as strings), and `null`.
+	 * Other supported values are: component instances, functions, and `null`.
 	 *
 	 * The `size` value, as well as any other additional props, will be passed through.
 	 * @default null
 	 */
 	icon?: IconType | null;
 	/**
 	 * The size (width and height) of the icon.
-	 *
-	 * Defaults to `20` when `icon` is a string (i.e. a Dashicon id), otherwise `24`.
-	 * @default `'string' === typeof icon ? 20 : 24`.
+	 * @default 24
 	 */
 	size?: number;
 } & AdditionalProps< IconType >;
```

Comment:
In #102668 I had to add `className` to the list of accepted props, as it was needed when applying styles via CSS modules. Although it's not strictly necessary for now

Reply:
Possibly because you chose `Record< string, never >` [here](https://github.com/Automattic/wp-calypso/pull/102668/files#diff-99714f616876923d817e7694b83c50786b0778ee9a87fe7ef61bf50b97c5d41cR13).

I opted for a looser `Record< string, unknown >`, which I thought was maybe closer to the original outcome (if not the original intent) of `{}`.